### PR TITLE
351 finality provider list UI

### PR DESCRIPTION
**Pull request by jeremy-babylonlabs**
ss-1
ss-2
ss-3
ss-4
ss-5

### CODE REVIEW

**Comment by jeremy-babylonlabs** [comment]
added fp list, pending a second look on https://github.com/babylonlabs-io/bbn-core-ui/pull/50
will double check and make this PR ready once that is merged.

File: src/app/components/Staking/FinalityProviders/FinalityProviderSubtitle.tsx
Comment (totraev) [inline src/app/components/Staking/FinalityProviders/FinalityProviderSubtitle.tsx:1]: Do we really need this component?

File: src/app/components/Staking/FinalityProviders/FinalityProviderTable.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          Number(a.commission) - Number(b.commission),
+      },
+      {
+        key: "state",
+        header: "Status",
+        render: (value: unknown) => mapStatus(value as FinalityProviderState),
+      },
+    ],
+    [coinName],
+  );
```
Comment (totraev) [inline src/app/components/Staking/FinalityProviders/FinalityProviderTable.tsx:90]: move outside the rendering function

File: src/app/components/Staking/FinalityProviders/FinalityProviderTable.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    finalityProviders,
+    hasNextPage,
+    fetchNextPage,
+    searchValue,
+    filterValue,
+    hasError,
+    handleRowSelect,
+  } = useFinalityProviderService();
+
+  const { coinName } = getNetworkConfig();
```
Comment (totraev) [inline src/app/components/Staking/FinalityProviders/FinalityProviderTable.tsx:31]: move it outside the rendering function

File: src/app/components/Staking/FinalityProviders/FinalityProviderTable.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    ],
+    [coinName],
+  );
+
+  const tableData = useMemo(
+    () => finalityProviders?.map((fp) => ({ ...fp, id: fp.btcPk })) || [],
+    [finalityProviders],
+  );
+
+  const renderContent = () => {
```
Comment (totraev) [inline src/app/components/Staking/FinalityProviders/FinalityProviderTable.tsx:97]: You don't need this function

File: src/app/components/Staking/FinalityProviders/FinalityProviderTitle.tsx
Comment (totraev) [inline src/app/components/Staking/FinalityProviders/FinalityProviderTitle.tsx:1]: What is the purpose of this component?

File: src/app/hooks/services/useFinalityProviderService.ts [hunk 4]
```diff
@@ -14,18 +17,36 @@ const SORT_DIRECTIONS = {
   asc: undefined,
 } as const;
 
-export function useFinalityProviderService() {
+export function useFinalityProviderServiceState() {
```
Comment (totraev) [inline src/app/hooks/services/useFinalityProviderService.ts:20]: ServiceState ?

File: src/app/state/FinalityProviderServiceState.tsx
Comment (totraev) [inline src/app/state/FinalityProviderServiceState.tsx:1]: I don't understand this

**Review by totraev — CHANGES_REQUESTED** [review]
I'll need update this PR later

File: src/app/state/FinalityProviderServiceState.tsx
Comment (jeremy-babylonlabs) [reply src/app/state/FinalityProviderServiceState.tsx:1]: my bad, my updated implementation was weirdly not included, resolved

**Review by totraev — APPROVED** [review]

File: src/app/state/FinalityProviderState.tsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+      });
+    } catch (error: any) {
+      showError({
+        error: {
+          message: error.message,
+          errorState: ErrorState.SERVER_ERROR,
+        },
+      });
+      captureError(error);
+      return [];
```
Comment (totraev) [inline src/app/state/FinalityProviderState.tsx:151]: side effect inside useMemo function

File: src/app/state/FinalityProviderState.tsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  const handleFilter = useCallback((value: string | number) => {
+    setFilterValue(value);
+  }, []);
+
+  const handleRowSelect = useCallback((row: FinalityProvider) => {
+    setSelectedFinalityProvider(row);
+  }, []);
+
+  const filteredFinalityProviders = useMemo(() => {
+    try {
```
Comment (totraev) [inline src/app/state/FinalityProviderState.tsx:117]: why we need try/catch here?

**Review by totraev — APPROVED** [review]

**Pull request merged**